pkg/prometheus/promcfg.go

```diff
@@ -2229,14 +2227,9 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 
 func (cg *ConfigGenerator) generateRoleSelectorConfig(k8sSDConfig yaml.MapSlice, roles []string, selector metav1.LabelSelector) yaml.MapSlice {
 	selectors := make([]yaml.MapSlice, 0, len(roles))
-	labelSelector := labels.SelectorFromValidatedSet(labels.Set(selector.MatchLabels))
-
-	for _, exp := range selector.MatchExpressions {
-		requirement, err := labels.NewRequirement(exp.Key, selection.Operator(strings.ToLower(string(exp.Operator))), exp.Values)
-		if err != nil {
-			panic(fmt.Errorf("failed to create label requirement: %w", err))
-		}
-		labelSelector = labelSelector.Add(*requirement)
+	labelSelector, err := metav1.LabelSelectorAsSelector(&selector)
+	if err != nil {
+		panic(fmt.Errorf("failed to convert label selector to selector: %w", err))
```

Comment:
(nit) while we're at it, maybe add a comment justifying the panic?
```suggestion
		// The field must have been validated by the controller beforehand.
		// If we fail here, it's a functional bug.
		panic(fmt.Errorf("failed to convert label selector to selector: %w", err))
```